# HG changeset patch
# User Clemens Lang <cal@macports.org>
# Date 1449516538 -3600
#      Mon Dec 07 20:28:58 2015 +0100
# Node ID c5694b7d4b8f5acd857ad4681a29ca79c76d1e59
# Parent  30679b45eeadb1ece557963cbd5915ab933fa747
Support long(er) folder names in sidebar

Folder names longer than 31 characters were previously cut off in the
sidebar, even though the field provided space for 128 in the current
implementation. Change to a generic sizeof() expression that will grow
with the size of the declaration.

Signed-off-by: Clemens Lang <cal@macports.org>
License: GPL-2.0+
Upstream-Status: Submitted [private email 2015-12-07]

diff --git a/sidebar.c b/sidebar.c
--- a/sidebar.c
+++ b/sidebar.c
@@ -157,7 +157,7 @@
     sbe.new = new;
     sbe.flagged = flagged;
     sbe.size = size;
-    strncpy(sbe.box, box, 31);
+    strncpy(sbe.box, box, sizeof(sbe.box) - 1);
 
     safe_realloc(&entry, SBvisual + 2);
     entry[SBvisual + 1] = '\0';
